### ISSUE TITLE
Label: `angle` prop is rejected by TypeScript

### ISSUE THREAD

**Issue by eps1lon** [recharts/recharts#2400]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.
  A fix for https://github.com/recharts/recharts/issues/2128 was already released but does not fix `angle` on `Label`

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/restless-paper-4u53c?file=/src/Dashboard.tsx)

### Steps to reproduce
```tsx
<YAxis stroke="black">
  <Label
    angle={270}
    position="left"
    style={{
      textAnchor: 'middle',
      fill: "black",
    }}
  >
    Sales ($)
  </Label>
</YAxis>
```

### What is expected?
`angle` is accepted by TypeScript.

### What is actually happening?
"Property `angle` does not exist on type ..."

The rendered label is rotated. This is purely an issue at the type level.

| Environment | Info |
|---|---|
| Recharts | v2.0.3 |
| React | 16.14.0 |
| System | Ubuntu 20.04 |
| Browser | Chrome 87 |

Update: Use real-world example by default since an isolated usage of `<Label />` was confusing.

**Comment by dillonreedy** [comment]
angle isn't a prop on a `<Label/>` component, it is a prop on a `<Text/>` component.

I've made a codesandbox for you here using the angle prop on a Text component:
[![Edit serene-bush-b1vdj](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/serene-bush-b1vdj?fontsize=14&hidenavigation=1&theme=dark)

**Comment by eps1lon** [comment]
> angle isn't a prop on a <Label/> component, it is a prop on a <Text/> component.

Are you saying that `angle` has not effect on a `<Label />`? If I change it from `270` to `0` in https://codesandbox.io/s/practical-bhaskara-yvuvs?file=/src/Dashboard.js then the label rotates as expected.

**Comment by dillonreedy** [comment]
If you use the label component inside axis component, then you can control its  rotation through the angle prop.

A standalone label component has no interface to an angle prop though. You'll instead want to use a Text component if you're attempting to create some kind of standalone text.

**Comment by eps1lon** [comment]
> A standalone label component has no interface to an angle prop though. You'll instead want to use a Text component if you're attempting to create some kind of standalone text.

I don't want to use a `Text` component since I'm using the `Label` inside an axis component. The `Label` inside an `Axis` does implement the `angle` prop but TypeScript is rejecting that prop. Does this clarify the issue for you?

**Comment by TiboC** [comment]
I have the same issue here.

**Comment by nicolajkirchhof** [comment]
Me too. Currently I'm just ignoring the error with '// @ts-ignore' and are happy that its still works as expected.

**Comment by xile611** [comment]
Fixed in v2.0.7

**Comment by oliviertassinari** [comment]
Fix validated in https://github.com/recharts/recharts/issues/2400
